src/AzureAD.tsx

```diff
@@ -94,12 +94,14 @@ class AzureAD extends React.Component<IProps, IState> {
   }
 
   public updateAuthenticationState = (state: AuthenticationState, user?: IAccountInfo) => {
+    if (user) {
+      this.dispatchToProvidedReduxStore(user);
+    }
     this.setState({
       authenticationState: state
     });
 
     if (user) {
```

Comment:
It seems like maybe this block should only execute after the `authenticationState` is guaranteed to have been set. I would probably move it into the setState callback that gets executed only after the asynchronous state update has been set.

Reply:
You are totally right.
Since it is not a big change i will try to find tonight some time (cannot promise though) and push it with your suggestions.